CMakeLists.txt

```diff
@@ -334,6 +334,18 @@ if (USE_OPENSSL)
     message(FATAL_ERROR "TimescaleDB requires OpenSSL version 1.0 or greater")
   endif ()
 
+  if(MSVC)
+    set(_libraries)
+    foreach(_path ${OPENSSL_LIBRARIES})
+      get_filename_component(_dir ${_path} DIRECTORY)
+      get_filename_component(_name ${_path} NAME_WE)
+      string(REGEX REPLACE "[Dd]$" "" _fixed ${_name})
```

Comment:
I guess this means we will never use debug binaries. Is this really what we want?

Reply:
Two options:

1. We can only replace the debug library with the release library if the debug library is missing.
2. We can have an option that prevent the rewrite.

Option 1 would not generate an error if the debug binaries should be used but do not exists, which might be desirable in some cases. Option 2 would handle that, but would require the option to be explicitly set if debug binaries are desired.

Reply:
I guess I am not at all following what we're trying to achieve here. Logically, if you are attempting a debug build and you don't have all dependencies installed, it seems natural that the build should fail until you install the missing dependencies. Otherwise you should use the release build.

I guess I am not sure why we're tying ourselves in knots to circumvent missing libraries in order to build a hybrid. 

Isn't this simply about displaying better errors, such as:

"OpenSSL debug libraries not found. Install the debug libraries or try a release build instead".

Reply:
Originally this started as the fact that our builds were failing to install for customers who were using enterpriseDB on windows, because of naming convention changes for the SSL libraries. I discovered this issue while debugging that. Incidentally, I believe this will fix both issues.

Reply:
In this case, the user has installed EnterpriseDB and want to do a debug build of TimescaleDB but that generates strange errors because the debug libraries are not included in EnterpriseDB.  Installing packages on Windows in, in general, quite a nightmare and allowing developers to build TimescaleDB with the EnterpriseDB installed is something that we should support.

It will, of course, not allow debugging the OpenSSL code, but I think that is a minor issue.

Reply:
FWIW, I have concerns about adding these kinds one-off "hacks" to our build system and code, especially on a platform we don't really support for development. Where do we draw the line? AFAIK, nothing is really wrong without this fix, although errors can certainly be improved. And, I don't think we're in the business of making Windows easier to use. Just saying. 

The work is done now, but please add a comment on why this hack is needed because now it isn't obvious. Also not that for people that does have debug binaries installed, they won't be used anymore.